src/server/visual-diff-plugin.js

```diff
@@ -114,6 +115,21 @@ function extractTestPartsFromName(name) {
 	};
 }
 
```

Comment:
This is the code that's saving additional information about each test so that the reporter can ask for it later. As you can see, because there are so many return paths, having to set it from 4 different places isn't ideal. Hopefully some future refactoring can resolve this!